src/re/correlated_field.py

```diff
@@ -65,67 +157,79 @@ def get_fourier_mode_distributor(
             m_length = jnp.expand_dims(m_length, axis=-1) + tmp
         m_length = jnp.sqrt(m_length)
 
-    # Construct an array of unique mode lengths
-    uniqueness_rtol = 1e-12
-    um = jnp.unique(m_length)
-    tol = uniqueness_rtol * um[-1]
-    um = um[jnp.diff(jnp.append(um, 2 * um[-1])) > tol]
-    # Group modes based on their length and store the result as power
-    # distributor
-    binbounds = 0.5 * (um[:-1] + um[1:])
-    m_length_idx = jnp.searchsorted(binbounds, m_length)
-    m_count = jnp.bincount(m_length_idx.ravel(), minlength=um.size)
-    if jnp.any(m_count == 0) or um.shape != m_count.shape:
-        raise RuntimeError("invalid harmonic mode(s) encountered")
-
-    return m_length_idx, um, m_count
+    return _unique_mode_distributor(m_length)
 
 
 RegularCartesianGrid = namedtuple(
```

Comment:
I agree on all points but I think all three of them can also be implemented at a later point :)

* Agreed, ideally grids should not get inlined into computations. So far I opted for simplicity but I assume there exists a simple solution now with jft.Model which would allow us pulling grids out of the model.
* Personally, I like the "hacky" implementation (ab)using `shape` to store the `nside` but I am happy to explicitly allow grids. Technically they are already exposed and can be manually instantiated. To pass them to `add_fluctuations`, we could make it such that shape can optionally be a grid, in which case `_make_grid` is skipped. We could also create a second `add_fluctuations_with_grid` that directly takes the grid and uses it in `add_fluctuations`.
* Yes, we should :see_no_evil:

Reply:
Awesome! Great to see us on the same page here. I opened a corresponding enhancement issue #26 to revisit this at a later point.